Customize Conduit's built-in connectors (#105)
ConduitIO/conduit-site: docs/connectors/additional-built-in-plugins.mdx

---

> title: "Add Built-in Connectors"

Reviewer: Other pages are named using the infinitive verb forms (installing, referencing, building...) so my suggestion is to use this form here as well.

```suggestion
title: "Adding built-in Connectors"
```

sidebar_position: 2
---

Built-in connectors offer better performance when compared to standalone ones,
which is why in some cases it's desirable to have a custom build of Conduit that
includes additional built-in connectors.

The simplest way to achieve so is to write a small application that embeds
Conduit (i.e. uses Conduit as a library) and adds one or more connectors to its
default configuration.

In the example below we will add
the [HTTP connector](https://github.com/conduitio-labs/conduit-connector-http)
to Conduit as a built-in connector.

First, we initialize a Go module with `go mod init github.com/conduitio-labs/custom-conduit`.

Then , we need to add Conduit and the HTTP connector as dependencies:
```shell
go get github.com/conduitio/conduit
go get github.com/conduitio-labs/conduit-connector-http
go mod tidy
```

Once that is done, we need to write a `main` function that:

> 1. adds the HTTP connector the default Conduit configuration

Reviewer: ```suggestion
1. Adds the HTTP connector the default Conduit configuration
```

> 2. runs Conduit with the custom configuration.

Reviewer: ```suggestion
2. Runs Conduit with the custom configuration.
```

That's done in the code below:

```go
package main

import (
	http "github.com/conduitio-labs/conduit-connector-http"
	"github.com/conduitio/conduit/pkg/conduit"
)

func main() {
	// Get the default configuration, including all built-in connectors
	cfg := conduit.DefaultConfig()

	// Add the HTTP connector to list of built-in connectors
	cfg.ConnectorPlugins["http"] = http.Connector

	conduit.Serve(cfg)
}
```

> The "custom Conduit" can be built with `go build main.go -o custom-conduit`. If

Reviewer: ```suggestion
This custom version of Conduit can be built with `go build main.go -o custom-conduit`. If
```
